pkg/apis/stork/v1alpha1/types.go

```diff
@@ -308,3 +315,253 @@ const (
+	// ClusterName name of the storage cluster
+	ClusterName string `json:"clusterName"`
+	// ClusterUUID uuid for the storage cluster
+	ClusterUUID string `json:"clusterUuid"`
+	// CreatedAt timestamp at which the storage cluster was created
+	CreatedAt *meta.Time `json:"createdAt"`
+	// Status status of the storage cluster
+	Status ClusterStatus `json:"status"`
+	// Reason is human readable message indicating the status of the cluster
+	Reason string `json:"reason,omitempty"`
+	// NodeStatuses list of statuses for all the nodes in the storage cluster
+	NodeStatuses []NodeStatus `json:nodes`
+}
+
+// ClusterStatus is the enum type for cluster statuses
+type ClusterStatus string
+
+// These are valid cluster statuses.
+const (
+	// ClusterOK means the cluster is up and healthy
+	ClusterOk ClusterStatus = "Ok"
+	// ClusterOffline means the cluster is offline
+	ClusterOffline ClusterStatus = "Offline"
+	// ClusterNotInQuorum means the cluster is out of quorum
+	ClusterNotInQuorum ClusterStatus = "NotInQuorum"
+	// ClusterUnknown means the cluser status is not known
+	ClusterUnknown ClusterStatus = "Unknown"
+)
+
+// NodeStatus status of the storage cluster node
+type NodeStatus struct {
+	// NodeName name of the node
+	NodeName string `json:"nodeName"`
+	// NodeUUID uuid of the node
+	NodeUUID string `json:"nodeUuid"`
+	// Network details used by the storage driver
+	Network NetworkStatus `json:"network"`
+	// Geo topology information for a node
+	Geo Geography `json:"geography"`
+	// Conditions is an array of current node conditions
+	Conditions []NodeCondition `json:"conditions,omitempty"`
+}
+
+// NetworkStatus network status of the node
+type NetworkStatus struct {
+	// DataIP is the IP address used by storage driver for data traffic
+	DataIP string `json:"dataIP"`
+	// MgmtIP is the IP address used by storage driver for management traffic
+	MgmtIP string `json:"mgmtIP"`
+}
+
+// NodeCondition contains condition information for a node
+type NodeCondition struct {
+	// Type of the node condition
+	Type NodeConditionType `json:"type"`
+	// Status of the condition
+	Status ConditionStatus `json:"status"`
+	// Reason is human readable message indicating details about the condition status
+	Reason string `json:"reason"`
+}
+
+// NodeConditionType is the enum type for different node conditions
+type NodeConditionType string
+
+// These are valid conditions of the storage node. They correspond to different
+// components in the storage cluster node.
+const (
+	// NodeState is used for overall state of the node
+	NodeState NodeConditionType = "NodeState"
+	// StorageState is used for the state of storage in the node
+	StorageState NodeConditionType = "StorageState"
+)
+
+// ConditionStatus is the enum type for node condition statuses
+type ConditionStatus string
+
+// These are valid statuses of different node conditions.
+const (
+	// NodeOnline means the node condition is online and healthy
+	NodeOnline ConditionStatus = "Online"
```

Comment:
Can be made in a subsequent change but these should be called `ConditionOnline` etc.
Also need more Statuses like `StorageFull`, `DiskError`, etc